chemprop/cli/train.py

```diff
@@ -551,6 +563,25 @@ def main(args):
         raise ArgumentError(
             argument=None, message="'val_path' must be specified if 'test_path' is provided!"
         )  # TODO: In v1 this wasn't the case?
+    
+    if args.save_smiles_splits:
+        if multicomponent:
+            train_smis = [[Chem.MolToSmiles(data.mol) for data in dset] for dset in train_data]
```

Comment:
[datasets ](https://github.com/chemprop/chemprop/blob/v2/dev/chemprop/data/datasets.py#L174) have `.smiles` as an attribute. I recommend using that to avoid duplicating code.

Reply:
Thanks for the suggestion! I've modified this.